components/engine/src/rocks/util/engine_metrics.rs

```diff
@@ -418,471 +443,230 @@ pub fn flush_engine_ticker_metrics(t: TickerType, value: u64, name: &str) {
+                .inc_by(v);
+        }
+        TickerType::BlobDbNumPrev => {
+            STORE_ENGINE_BLOB_LOCATE_VEC
+                .with_label_values(&[name, "number_blob_prev"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbNumKeysWritten => {
+            STORE_ENGINE_BLOB_FLOW_VEC
+                .with_label_values(&[name, "keys_written"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbNumKeysRead => {
+            STORE_ENGINE_BLOB_FLOW_VEC
+                .with_label_values(&[name, "keys_read"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbBlobFileBytesWritten => {
+            STORE_ENGINE_BLOB_FLOW_VEC
+                .with_label_values(&[name, "bytes_written"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbBlobFileBytesRead => {
+            STORE_ENGINE_BLOB_FLOW_VEC
+                .with_label_values(&[name, "bytes_read"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbBlobFileSynced => {
+            STORE_ENGINE_BLOB_FILE_SYNCED
+                .with_label_values(&[name, "file_sync"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcNumFiles => {
+            STORE_ENGINE_BLOB_GC_FILE_VEC
+                .with_label_values(&[name, "blob_db_num_seek"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcNumNewFiles => {
+            STORE_ENGINE_BLOB_GC_FILE_VEC
+                .with_label_values(&[name, "blob_db_num_seek"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbBytesWritten => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "bytes_written"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbBytesRead => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "bytes_read"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcNumKeysOverwritten => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "keys_overwritten"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcNumKeysRelocated => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "keys_relocated"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcBytesOverwritten => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "bytes_overwritten"])
+                .inc_by(v);
+        }
+        TickerType::BlobDbGcBytesRelocated => {
+            STORE_ENGINE_BLOB_GC_FLOW_VEC
+                .with_label_values(&[name, "bytes_relocated"])
+                .inc_by(v);
+        }
         _ => {}
     }
 }
 
+macro_rules! engine_histogram_metrics {
+    ($name:ident, $db:expr, $value:expr) => {
+        concat_idents!(STORE_ENGINE_, $name, _VEC)
+            .with_label_values(&[$db, concat!(stringify!(lower($name)), "_median")])
```

Comment:
```suggestion
            .with_label_values(&[$db, concat!(stringify!(lower!($name)), "_median")])
```

but as mentioned above, we don't really need `casey`.